packages/connect-msw/src/create-worker-handlers.ts

```diff
@@ -0,0 +1,123 @@
+import { MethodIdempotency } from "@bufbuild/protobuf";
+import type {
+  CallOptions,
+  ConnectRouter,
+  HandlerContext,
+} from "@connectrpc/connect";
+import type { UniversalHandler } from "@connectrpc/connect/protocol";
+import { http, HttpResponse, passthrough as mswPassthrough } from "msw";
+import type { RequestHandler, ResponseResolver } from "msw";
+
+const passthroughHeader = "x-passthrough";
+
+export function passthrough(ctx: HandlerContext) {
+  ctx.responseHeader.append(passthroughHeader, "true");
+  return {};
+}
+
+export function createBypassOptions(callOptions: CallOptions): CallOptions {
+  return {
+    ...callOptions,
+    headers: {
+      // We could make this more consistent with the exposed bypass() function provided by msw
+      // IFF we could specify interceptors on a request by request basis. At least in theory, bypass()
+      // works on a standard Request object so we'd need a converter from the interceptor UnaryRequest/StreamRequest
+      // to a standard request to pass it to bypass() and out again.
+      // Otherwise, we just need to make sure this keeps in sync with https://github.com/mswjs/msw/blob/8855956f561ad4afd4ce7ae6500ca706757bb5a9/src/core/bypass.ts
+      "x-msw-intention": "bypass",
```

Comment:
@timostamm wanted to get your thoughts on this way of adding headers to a bunch of requests. Basically this header exists to tell msw to have a request bypass all mock handlers. This can be applied to any client request (as opposed to passthrough, which is designed to just the incoming request). It would be nice to have a way to intercept the Request before it gets sent to fetch so we can more seamlessly fit with the `bypass()` function.

Not really blocking, just wanted to raise it.

Reply:
What is bypass for? I see this comment in the source:

>  Unlike "passthrough()", bypass is meant for performing additional requests within pending request resolution.

If I intercept a request, and need to make some requests to provide a mocked response, do I need to set this header to prevent MSW from intercepting these requests as well?

That's pretty straight-forward with:

```ts
client.say({sentence: "hi"}, { headers: { "x-msw-intention": "bypass" } });
```

Users could also set up a transport specifically for this use case, with a connect interceptor that always sets this header. 

Seems to me that this just needs good examples in our documentation, but please let me know if I'm missing something. 

MSW might be able to provide slightly convenience by providing a `fetch` function to response resolvers that always sets this header, and potentially inherits headers and cookies - similar to Svelte.

Reply:
> If I intercept a request, and need to make some requests to provide a mocked response, do I need to set this header to prevent MSW from intercepting these requests as well?

Yes that is correct. 

> Seems to me that this just needs good examples in our documentation, but please let me know if I'm missing something.

Yeah maybe docs are sufficient. I wanted to raise it to see if there was any feature requests to allow interceptors on a per request basis. I've pushed up a custom fetch that can be passed. It's a little cleaner than a custom interceptor and can still be wrapped if someone needs an additional custom fetch.

Reply:
>  I wanted to raise it to see if there was any feature requests to allow interceptors on a per request basis.

Do you mean connect interceptors as described [here](https://connectrpc.com/docs/web/interceptors)? I think it would be a bad design choice to allow the per request because it would soften the separation of concerns, and opens up questions about order of interceptors. If you want to run interceptors conditionally, https://github.com/connectrpc/connect-es/pull/841 is the way to go.

Reply:
Ah yeah the lack of clarity on ordering makes sense. An easy way to work around that would be to instead allow a custom fetch to be defined on a per request basis, which wouldn't introduce any ordering confusion. But for now, a custom transport to handle bypasses will suffice. I don't think it's worth much api modification just to smooth out things like msw bypass.